src/engine_render.c

```diff
@@ -5363,9 +5363,70 @@ static void display_fast_drawlist(struct Camera *cam)
     }
 }
 
-static long convert_world_coord_to_front_view_screen_coord(struct Coord3d* pos, struct Camera* cam, long* x, long* y, long* z)
+// Returns 1 if projected point is withing player's window, 0 otherwise
+static long project_point_helper(struct PlayerInfo *player, int zoom, int a1, int a2, int a3, unsigned short pos_z, long *x_out, long *y_out, long *z_out)
 {
-    return _DK_convert_world_coord_to_front_view_screen_coord(pos, cam, x, y, z);
+    int v11;
+    int64_t v13;
+    uint8_t v13_byte4;
+    short window_width = player->engine_window_width;
+    short window_height = player->engine_window_height;
+
+    *x_out = (zoom * a2 >> 16) + (*(uint16_t *)&window_width >> 1);
+    v11 = zoom * (a1 - a3) >> 8;
+    *z_out = window_height - ((v11 + ((uint16_t)(window_height & 0xFFFE) << 7)) >> 8) + 64;
+    v13 = zoom * *(int16_t *)&pos_z << 7;
+    v13_byte4 = *((uint8_t *)&v13 + 4);
+    *y_out = (v11 + ((uint16_t)(window_height & 0xFFFE) << 7) - ((v13_byte4 + (signed int)v13) >> 16)) >> 8;
+
+    return (*x_out >= 0 && *x_out < window_width && *y_out >= 0 && *y_out < window_height);
+}
+
+static long convert_world_coord_to_front_view_screen_coord(struct Coord3d* pos, struct Camera* cam, long* x_out, long* y_out, long* z_out)
+{
+    int zoom;   // TODO: Change this to correct name
+    unsigned int orientation;
+    int a1, a2, a3;
+    long result = 0;
+    struct PlayerInfo* player = get_my_player();
+
+    // return _DK_convert_world_coord_to_front_view_screen_coord(pos, cam, x_out, y_out, z_out);
+
+    zoom = 32 * cam->zoom / 256;
+    orientation = ((unsigned int)(cam->orient_a + 256) >> 9) & 3;
+
+    switch ( orientation )
+    {
+        case 0:
```

Comment:
I think there's already an enum for 'directions', you could look into using that or making your own enum.